src/main/tut/2.2-limits-and-colimits.md

```diff
@@ -0,0 +1,275 @@
+```Haskell
+p' = p . m
+q' = q . m
+```
+```scala
+val p' = p compose m
+val q' = q compose m
+
+```
+...................
+
+```Haskell
+f :: c' -> c
+```
+```scala
+val f : C' => C
+```
+..................
+```Haskell
+u :: c -> Lim D
+```
+```scala
+val u :  C => Lim[D]
+```
+..................
+```Haskell
+u . f :: c' -> Lim D
+```
+```scala
+u compose f : C' => Lim[D]
+```
+..................
+```Haskell
+contramap :: (c' -> c) -> (c -> Lim D) -> (c' -> Lim D)
+contramap f u = u . f
+```
+```scala
+def contramap[C, C', D[_]](f: C' => C, u: C => Lim[D]): (C' => Lim[D]) = 
+  u compose f
+```
+..................
+```Haskell
+f :: c' -> c
+```
+```scala
+val f : C' => C
+```
+..................
+```Haskell
+Nat(Δc, D) -> Nat(Δc', D)
+```
+```scala
+(Δc ~> D) => (Δc' ~> D)
+```
+..................
+```Haskell
+αa :: Δca -> D a
+```
+```scala
+val αA : ΔcA => D[A]
+```
+..................
+```Haskell
+αa :: c -> D a
+```
+```scala
+val αA : C => D[A]
+```
+..................
+```Haskell
+βA :: c' -> D A
+```
+```scala
+val βA : C' => D[A]
+```
+..................
+```Haskell
+βa = αa . f
+```
+```scala
+val βA = αA compose f 
+```
+..................
+```Haskell
+c -> Nat(Δc, D)
+```
+```scala
+C => (Δc ~> D) 
```

Comment:
translation of `Nat` (natural transformation) needs to be consistent with previous sections.